soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -210,9 +228,9 @@ __weak void pm_state_set(enum pm_state state, uint8_t substate_id)
 		z_xt_ints_off(0xffffffff);
 		core_desc[cpu].bctl = DSPCS.bootctl[cpu].bctl;
 		DSPCS.bootctl[cpu].bctl &= ~DSPBR_BCTL_WAITIPCG;
-		soc_cpus_active[cpu] = false;
-		z_xtensa_cache_flush_inv_all();
 		if (cpu == 0) {
```

Comment:
Why did you move this if ? Don't you need `soc_cpus_active[cpu] = false;` for cores != 0 too ?

Reply:
For secondary core its done in _**power_gate_entry**_ function.